Packages/com.unity.inputsystem/Tests/InputSystem/CoreTests_Devices.cs

```diff
@@ -3369,4 +3369,90 @@ public void Devices_CanUseNoiseFilterWithMultipleProcessors()
 
         Assert.AreEqual(NoisyInputDevice.current, device1);
     }
+
+    [Test]
+    [Category("Devices")]
+    public void Devices_IMECompositionEventsGoThroughKeyboard()
```

Comment:
These tests all show a focus on "XYZ works". This test for example, focuses on a mechanic that is irrelevant to the user. Same for `Devices_IMEEnableSendsCorrectIOCTLCommand`, for example.

Each test should broadly go in one of three categories:
- Highlight something that *can* be done (`Devices_CanBeReadded`)
- Highlight something that *cannot* be done (`Devices_CannotChangeSetupOfDeviceWhileAddedToSystem`)
- Highlight a notable behavior when something *is* done (`Devices_WhenCreationFails_SystemRecoversGracefully`)

I recommend as thinking of the verification aspect of tests entirely as a side effect and focusing entirely on the "this is how I do" aspect in tests.